scala-package/core/src/main/scala/ml/dmlc/mxnet/Optimizer.scala

```diff
@@ -71,6 +119,10 @@ abstract class Optimizer extends Serializable {
   // Dispose the state it created
   def disposeState(state: AnyRef): Unit
 
+  def serializeState(state: AnyRef): Array[Byte]
+
+  def deserializeState(bytes: Array[Byte]): AnyRef
```

Comment:
my opinion is that AnyRef force the user to call asInstanceOf[AnyRef], or asInstanceOf[something] every time...

is it possible to introduce a class framework for the user to implement `State`, e.g. they have to implement some method giving the binary representation of their implementation, so that we can call them directly here?

yeah, you are right, any framework introduces overhead to impl